COLLABORATOR_GUIDE.md

```diff
@@ -364,13 +364,13 @@ there is expected impact on the user community.
 
 Node.js uses three Deprecation levels:
 
-* *Documentation-Only Deprecation* refers to elements of the Public API that are
-  being staged for deprecation in a future Node.js major release. An explicit
-  notice indicating the deprecated status is added to the API documentation
-  but no functional changes are implemented in the code. There will be no
-  runtime deprecation warnings emitted for such deprecations by default.
+* *Documentation-Only Deprecation* refers to elements of the Public API that
+  should be avoided by developers. An explicit notice indicating the deprecation
+  status is added to the API documentation but no functional changes are
+  implemented in the code.
+  There will be no warnings emitted for such deprecations at runtime by default.
   Documentation-only deprecations may trigger a runtime warning when launched
-  with [`--pending-deprecation`][] flag (or its alternative,
+  with [`--pending-deprecation`][] flag (or its alternative, the
```

Comment:
This paragraph says that "Documentation-Only" is a deprecation level, however the "pending" implies that it has not been deprecated _yet_.
I do not think this terminology is consistent with the `--pending-deprecation` flag.

IMHO `--pending-deprecation` needs to be better explained as another level, something like "pending a runtime deprecation".

Reply:
@mcollina I don't think we can rename the flag anymore, although I agree that the name is unfortunate. I think the text is explicit enough about what the flag *does*, but I'm not sure we should codify that `--pending-deprecations` means that we are definitely going to runtime-deprecate something at some point?

(For context, this text is from the rather recent #18433, /cc @ChALkeR)

Reply:
I think we should name "pending runtime deprecation" as its own deprecation level. If not, we should try to define this better. Before this change, "doc-only deprecations" were a step towards "runtime deprecations" (in most cases). after this change, they became their own thing, which not necessarily would lead to a more severe deprecation. I think there is a space for something in between, were a doc-only deprecation would lead to a runtime deprecation.

Reply:
@mcollina See #18417.

I don't think there needs to be yet another separate deprecation level, in fact I think that most of the new doc-deprecations should come with `--pending-deprecation` support (where possible).

Reply:
> I'm not sure we should codify that --pending-deprecations means that we are definitely going to runtime-deprecate something at some point?

That's what it means though, doesn't it?  We currently use it for unsafe `new Buffer()` calls and I believe the plan is to warn more forcefully in the future.

Documentation-only deprecations are more for steering people towards the canonical solution.  `--pending-deprecations` doesn't need to warn about that because we're not going to runtime-deprecate it.

Reply:
documentation-only *could* but doesn't mean *must* be runtime deprecated. The only time it definitely should mean runtime deprecation is if we ultimately plan to end-of-life the code.

pending-deprecation *should* mean we'll eventually runtime deprecate.

Any documentation-only deprecation that we expect to move to runtime deprecation in the future should emit pending deprecation.